integration/raft/migration_test.go

```diff
@@ -869,3 +870,237 @@ func createDeliverEnvelope(n *nwo.Network, signer *nwo.SigningIdentity, blkNum u
+				orderers                               []*nwo.Orderer
+				peer                                   *nwo.Peer
+				channel                                string
+			)
+
+			BeforeEach(func() {
+				// create a channel with 3 orderers as consenters
+				orderer1 = network.Orderer("orderer1")
+				orderer2 = network.Orderer("orderer2")
+				orderer3 = network.Orderer("orderer3")
+				orderer4 = network.Orderer("orderer4")
+
+				orderers = []*nwo.Orderer{orderer1, orderer2, orderer3, orderer4}
+				consenters := []*nwo.Orderer{orderer1, orderer2, orderer3, orderer4}
+				peer = network.Peer("Org1", "peer0")
+				channel = "mychannel"
+				channelURL := fmt.Sprintf("/participation/v1/channels/%s", channel)
+
+				// starting orderers
+				for _, o := range orderers {
+					startOrderer(o)
+					cl := channelparticipation.List(network, o)
+					Expect(cl).To(Equal(channelparticipation.ChannelList{}))
+				}
+
+				genesisBlock := applicationChannelGenesisBlock(network, consenters, []*nwo.Peer{peer}, channel)
+				expectedChannelInfoPT := channelparticipation.ChannelInfo{
+					Name:              channel,
+					URL:               channelURL,
+					Status:            "active",
+					ConsensusRelation: "consenter",
+					Height:            1,
+				}
+
+				for _, o := range consenters {
+					By("joining " + o.Name + " to channel as a consenter")
+					channelparticipation.Join(network, o, channel, genesisBlock, expectedChannelInfoPT)
+					channelInfo := channelparticipation.ListOne(network, o, channel)
+					Expect(channelInfo).To(Equal(expectedChannelInfoPT))
+				}
+
+				for i, o := range orderers {
+					submitPeerTxn(o, peer, network, channelparticipation.ChannelInfo{
+						Name:              channel,
+						URL:               channelURL,
+						Status:            "active",
+						ConsensusRelation: "consenter",
+						Height:            uint64(1 + i), // since genesis block is already created.
+					})
+				}
+			})
+
+			AfterEach(func() {
+				for _, ordererProcess := range ordererProcesses {
+					ordererProcess.Signal(syscall.SIGTERM)
+					Eventually(ordererProcess.Wait(), network.EventuallyTimeout).Should(Receive())
+				}
+			})
+
+			Context("when updrading from raft to smartbft", func() {
+				var oldConsensusType, newConsensusType string
+
+				It("1) executes raft2smartbft green path", func() {
+					oldConsensusType = "etcdraft"
+					newConsensusType = "smartbft"
+
+					By("1) Config update on standard channel, State=MAINTENANCE, enter maintenance-mode")
+
+					getRaftMetaDataFromConfig := func() []byte {
+						config_ := nwo.GetConfig(network, peer, orderer1, channel)
+						consensusTypeValue_ := extractOrdererConsensusType(config_)
+						oldMetadata := consensusTypeValue_.Metadata
+						return oldMetadata
+					}
+
+					oldRaftMetadata := getRaftMetaDataFromConfig()
+
+					current, updated := prepareTransition(network, peer, orderer1, channel,
+						oldConsensusType, protosorderer.ConsensusType_STATE_NORMAL,
```

Comment:
are we doing both config changes in the same transaction?

1. Putting the channel to maintenance
2. Changing the consensus type

? 

We should first move to maintenance and then change the consensus type.